dts/bindings/adc/adc-comparator.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Comparator will monitor signal though ADC and based on configuration,
+    it will trigger a callback to user to inform signal criterion is met.
+
+    User is responsible for configuring comparator to establish threshold assert
+    value, comparison (less or equal, grater...) and callback handler.
+
+compatible: "adc-comparator"
+
+include: base.yaml
+
+properties:
+    io-channels:
+      type: phandle-array
+      required: true
+      description: |
+        ADC channel that will perform measurement.
+
+    threshold:
+      type: int
+      required: false
+      description: |
+        Value present on ADC as threshold assert.
```

Comment:
Units?

Reply:
Since this is yaml is meant to be extended only, I have been thinking of using millivolts rather than raw register value.
Probably have both.
- threshold-reg.
- threshold-mv.